packages/contracts-bedrock/src/L1/ETHLockbox.sol

```diff
@@ -0,0 +1,165 @@
+    event LockboxAuthorized(address indexed lockbox);
+
+    /// @notice Emitted when ETH liquidity is migrated from the current ETH lockbox to another.
+    /// @param lockbox The address of the ETH lockbox that was migrated.
+    event LiquidityMigrated(address indexed lockbox);
+
+    /// @notice Emitted when ETH liquidity is received during an authorized lockbox migration.
+    /// @param lockbox The address of the ETH lockbox that received the liquidity.
+    event LiquidityReceived(address indexed lockbox);
+
+    /// @notice The address of the SuperchainConfig contract.
+    ISuperchainConfig public superchainConfig;
+
+    /// @notice Mapping of authorized portals.
+    mapping(address => bool) public authorizedPortals;
+
+    /// @notice Mapping of authorized lockboxes.
+    mapping(address => bool) public authorizedLockboxes;
+
+    /// @notice Semantic version.
+    /// @custom:semver 0.0.1
+    function version() public view virtual returns (string memory) {
+        return "0.0.1";
+    }
+
+    /// @notice Constructs the ETHLockbox contract.
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice Initializer.
+    /// @param _superchainConfig The address of the SuperchainConfig contract.
+    /// @param _portals The addresses of the portals to authorize.
+    function initialize(address _superchainConfig, address[] calldata _portals) external initializer {
+        superchainConfig = ISuperchainConfig(_superchainConfig);
+        for (uint256 i; i < _portals.length; i++) {
+            _authorizePortal(_portals[i]);
+        }
+    }
+
+    /// @notice Authorizes a portal to lock and unlock ETH.
+    /// @param _portal The address of the portal to authorize.
+    function authorizePortal(address _portal) external {
+        if (msg.sender != PAO()) revert ETHLockbox_Unauthorized();
+        _authorizePortal(_portal);
+    }
+
+    /// @notice Getter for the current paused status.
+    function paused() public view returns (bool) {
+        return superchainConfig.paused();
+    }
+
+    /// @notice Receives the ETH liquidity migrated from an authorized lockbox.
+    function receiveLiquidity() external payable {
+        if (!authorizedLockboxes[msg.sender]) revert ETHLockbox_Unauthorized();
+        emit LiquidityReceived(msg.sender);
+    }
+
+    /// @notice Locks ETH in the lockbox.
+    ///         Called by an authorized portal on a deposit to lock the ETH value.
+    function lockETH() external payable {
+        if (!authorizedPortals[msg.sender]) revert ETHLockbox_Unauthorized();
+        emit ETHLocked(msg.sender, msg.value);
+    }
+
+    /// @notice Unlocks ETH from the lockbox.
+    ///         Called by an authorized portal when finalizing a withdrawal that requires ETH.
+    ///         Cannot be called if the lockbox is paused.
+    /// @param _value The amount of ETH to unlock.
+    function unlockETH(uint256 _value) external {
```

Comment:
Worth noting that this increases the cost of depositing and withdrawing, which means we should verify that the code inside of `CrossDomainMessenger` base gas function still provides enough gas.

Reply:
Ok nevermind, this should be completely fine since the XDM enforces this at a lower level and changes here don't impact the XDM execution